### ISSUE TITLE
Evaluate Tapjacking Dynamic Checks

### ISSUE THREAD

**Issue by cpholguera** [OWASP/owasp-mstg#1719]
In #1642 we've removed the "Dynamic Analysis" sub section from "### Testing for Overlay Attacks (MSTG-PLATFORM-9)". It should be researched carefully and we can include some POCs or similar. I tend to think that there's less that the app can do to avoid this, it's rather the system which has this vulnerability. So the tester can _only_ check the target/minSdk-value maybe and ensure that the OS version is not vulnerable. Of course if needed one can develop POCs such as https://github.com/iwo/marshmallow-tapjacking. 

Also the section included testing with QARK:

```
You can also test this vulnerability using an automated tool called [QARK](https://github.com/linkedin/qark "Qark"). [This article](https://resources.infosecinstitute.com/android-penetration-tools-walkthrough-series-qark "Android Penetration Tools Walkthrough Series: QARK") from Infosec institute explains this tool and how to use it for detecting tapjacking on OWASP's GoatDroid application.
```

The example is pretty outdated, the [exploit APK is built against SDK 18](https://github.com/linkedin/qark/blob/b6ad07b8956e5f056fbdac1bf777b06ed76a7260/qark/exploit_apk/app/src/main/AndroidManifest.xml#L18) and QARK seems to simply check the minSdk value being <9. See here: https://github.com/linkedin/qark/blob/b6ad07b8956e5f056fbdac1bf777b06ed76a7260/qark/plugins/manifest/min_sdk.py#L9

## Other comments to consider for this Issue

- https://github.com/OWASP/owasp-mstg/pull/1642/files#r392899486
- https://github.com/OWASP/owasp-mstg/pull/1642#issuecomment-586724090
- https://github.com/OWASP/owasp-mstg/pull/1642#issuecomment-625093954
- https://github.com/OWASP/owasp-mstg/pull/1642#issuecomment-625101963

## Note

The resulting PR might be simply: we really don't need a "Dynamic Analysis" sub section and we limit the testing to "verify the minSdk value".

**Comment by cpholguera** [comment]
This article is also interesting: https://www.xda-developers.com/how-tapjacking-made-a-return-with-android-marshmallow-and-nobody-noticed/

**Comment by jay0301** [comment]
> This article is also interesting: https://www.xda-developers.com/how-tapjacking-made-a-return-with-android-marshmallow-and-nobody-noticed/

@cpholguera Agreed on the point from Note that we can limit to verify minSDK value.

I was deep diving on dynamic testing and would agree that it is rather an AOSP issue than a application issue. Android versions from 8.0 and above has addressed these issues quite well. 

Below article gave me a more better idea. 

https://unit42.paloaltonetworks.com/unit42-android-toast-overlay-attack-cloak-and-dagger-with-no-permissions/

Definitely we can work on POCs if required. At present for exploitation or for generating proof of concepts below APKs would work. (Added one for link which has apk which creates multiple overlays on keyboard to capture keystroke. One of the exploits from cloak and dagger)

https://github.com/iwo/marshmallow-tapjacking
https://github.com/FSecureLABS/tapjacking-poc
https://github.com/DEVizzi/Invisible-Keyboard

I agree with the changes. What are your views on it shall I update the Dynamic Analysis section in issue #1642  to check for minSDK version and above links which would help the testers for proof of concept generation.

**Comment by cpholguera** [comment]
Hi @jay0301 , that's a very nice suggestion. thanks for the research, I agree with the minsdk. however that's more a static check, right? Maybe in dynamic analysis you can simply suggest taking those example POCs as a starting point. Please specify the Android version that they target with the format: "Android X.X (API level xx)"

**Comment by jay0301** [comment]
> Hi @jay0301 , that's a very nice suggestion. thanks for the research, I agree with the minsdk. however that's more a static check, right? Maybe in dynamic analysis you can simply suggest taking those example POCs as a starting point. Please specify the Android version that they target with the format: "Android X.X (API level xx)"

@cpholguera sure will do that. Drafting the section. Will share the commit today only for issue #1642 and include update Dynamic Analysis section.

**Comment by cpholguera** [comment]
perfect, thank you very much!

**Comment by cpholguera** [comment]
Closed with #1642
